src/coreclr/gc/gc.cpp

```diff
@@ -29124,6 +29307,13 @@ void gc_heap::compact_phase (int condemned_gen_number,
                     {
                         args.src_gennum = ((current_heap_segment == ephemeral_heap_segment) ? -1 : 2);
                     }
+                    if (heap_segment_mem (current_heap_segment) == heap_segment_allocated (current_heap_segment))
+                    {
+                        // the case of an empty region / segment - make sure
+                        // we don't compact anything, but go to the next region / segment
+                        assert (current_brick == end_brick);
+                        current_brick++;
+                    }
```

Comment:
same with this.

Reply:
I think my comments above apply to this location as well.